tests/test_vector2_truncate.py

```diff
@@ -12,12 +12,14 @@ def test_truncate_length(x: Vector2, max_length: float):
 
 
 @given(x=vectors(), max_length=lengths())
+@settings(suppress_health_check=[HealthCheck.filter_too_much])
```

Comment:
Is there a way to better handle generation rather than filtering so much?

Reply:
The main issue is that I don't think there's a reasonable way to generate `x` and `max_length` in such a way that `x.length <= max_length` without introducing a bunch of bias in the test.
I guess we could increase the range on `max_length` (or `x`, for the other test), which would reduce the likelihood of the assumption not being satisfied.

In any case, Hypothesis is able to generate enough samples, the issue is only that sometimes the heath check fails, resulting in a CI failure.

Reply:
PS: I tweaked the ranged to make constraint satisfaction more likely, going from ~49% to ~100% for `test_truncate_invariant`.

Before:
```
$ pytest --hypothesis-show-statistics --hypothesis-profile ci tests/test_vector2_truncate.py -v
[...]
tests/test_vector2_truncate.py::test_truncate_invariant:

  - 1000 passing examples, 0 failing examples, 1034 invalid examples
  - Typical runtimes: < 1ms
  - Fraction of time spent in data generation: ~ 60%
  - Stopped because settings.max_examples=1000
```

After:
```
$ pytest --hypothesis-show-statistics --hypothesis-profile ci tests/test_vector2_truncate.py -v
[...]
tests/test_vector2_truncate.py::test_truncate_invariant:

  - 1000 passing examples, 0 failing examples, 3 invalid examples
  - Typical runtimes: < 1ms
  - Fraction of time spent in data generation: ~ 61%
  - Stopped because settings.max_examples=1000

tests/test_vector2_truncate.py::test_truncate_equivalent_to_scale:

  - 1000 passing examples, 0 failing examples, 62 invalid examples
  - Typical runtimes: < 1ms
  - Fraction of time spent in data generation: ~ 58%
  - Stopped because settings.max_examples=1000
```